docs/decisions/backend/0010-advanced-installer-suggestions.md

```diff
@@ -0,0 +1,81 @@
+# Advanced Installer: Suggestions
+
+This document describes the design of the 'Suggestions' system within Advanced Installer, pictured
+in the following mockup below:
+
+![](./images/0009-advanced-installer-location.png)
+
+## Context Statement
+
+When using the Advanced Installer, the user should be provided with 'hints' dictating where a file might
+require to be placed based on a number of heuristics.
+
+This is part of the UX effort to 'Make Modding Easy' and a general requirement of our [Advanced Installer Design](./0009-advanced-installer-design.md).
+
+## Considered Options (Suggestions)
+
+- Reusing the Deployment System (`InstallFolderTarget`) for suggestions.
+- Creating a Suggestion System from scratch.
+
+## Decision Outcome
+
+Rather than splitting the metadata in two, we can leverage the existing
+`InstallFolderTarget` system in order to implement Advanced Installer suggestions;
+as that system already has required metadata.
+
+### Consequences
+
+- [Good] Strong code reuse as `InstallFolderTarget` already has required metadata to support this functionality.
+- [Neutral] Each `InstallFolderTarget` folder will now need a description.
+- [Neutral] All games will need to be converted to new `InstallFolderTarget` system.
```

Comment:
`InstallFolderTarget` was meant to be used by the Generic Folder installer to recognize folders.
If we unify the two things, we could have games that define InstallFolderTargets that are not actually recognizable.

So either we separate the two concepts with one that just wraps the other adding the information necessary for the Generic Installer to recognize the files, or we keep it a single thing but with a clear way for the Generic installer to know which InstallFolderTarget it should actually use (so perhaps two different methods on IGame returning the two different lists?)

Reply:
I'm not sure that I quite understand the problem here.

The initial set of 'suggested folders' (w/o suggestion algorithm) should consist of:
- All `GamePath` entries.
- All deploy-able folders not in `GamePath` (specified via `InstallFolderTarget`).

Are you suggesting that we might run into a situation where we might want to recommend subfolder(s) to the users which we may not necessarily want the Generic Installer to deploy to?

e.g.

```
Mod
├─ texture.dds
```

Where `texture.dds` should go to a folder like `{Game}/Data/Textures`, but `{Game}/Data/Textures` is not considered a folder under which the Generic Installer might try deploying to?

That doesn't happen often, but I'm happy to add an interative task to export the needed properties as an interface (trait), and expose another getter from `IGame` which provides custom suggestions that are not used as part of the deployment step. [And update the documentation accordingly]